cupyx/jit/_interface.py

```diff
@@ -138,7 +138,7 @@ def cached_codes(self):
             warnings.warn(
                 'No codes are cached because compilation is deferred until '
                 'the first function call.')
-        return [result.code for result in self._cached_codes]
+        return dict([(k, v.code) for k, v in self._cached_codes.items()])
```

Comment:
This fixes a bug introduced in #6575 IIUC. I can backport the fix to v10 if needed.

Reply:
Thanks for the reporting! Could you split the fix into another PR?

Reply:
Will do!

Reply:
See #6639.